tests/integration/conftest.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Fixtures for jubilant tests."""
+
+import pathlib
+
+import jubilant
+import pytest
+import yaml
+
+POSTGRESQL_CHANNEL = "14/stable"
+TEMPORAL_CHANNEL = "1.23/edge"
+
+
+@pytest.fixture(scope="module")
+def juju(request: pytest.FixtureRequest):
+    keep_models = bool(request.config.getoption("--keep-models"))
+
+    with jubilant.temp_model(keep=keep_models) as model:
+        model.wait_timeout = 10 * 60
+
+        yield model
+
+        if request.session.testsfailed:
+            log = model.debug_log(limit=1000)
+            print(log, end="")
+
+
+def deploy_temporal_stack(
+    juju: jubilant.Juju,
+    postgresql_channel: str = POSTGRESQL_CHANNEL,
+    temporal_channel: str = TEMPORAL_CHANNEL,
+    temporal_admin_channel: str = TEMPORAL_CHANNEL,
+):
+    """Deploy temporal-admin-k8s from the latest track.
+
+    Args:
+        juju: Juju object (jubilant)
+        postgresql_channel: channel of postgresql-k8s charm
+        temporal_channel: channel of temporal-k8s charm
+        temporal_admin_channel: channel of temporal-admin-k8s charm
+    """
+    juju.model_config(
+        values={
+            "update-status-hook-interval": "10s",
+        },
+    )
+
+    juju.deploy(
+        charm="postgresql-k8s",
+        app="postgresql-k8s",
+        channel=postgresql_channel,
+        trust=True,
+        base="ubuntu@22.04",
+    )
+
+    juju.deploy(
+        charm="temporal-k8s",
+        app="temporal-k8s",
+        channel=temporal_channel,
+        config={
+            "num-history-shards": 1,
+        },
+        base="ubuntu@22.04",
+    )
+
+    juju.deploy(
+        charm="temporal-admin-k8s",
+        app="temporal-admin-k8s",
+        channel=temporal_admin_channel,
+        base="ubuntu@22.04",
+    )
+
+    juju.integrate("temporal-k8s:db", "postgresql-k8s:database")
+    juju.integrate("temporal-k8s:visibility", "postgresql-k8s:database")
+
+    juju.integrate("temporal-k8s:admin", "temporal-admin-k8s:admin")
+
+    juju.wait(jubilant.all_active)
+
+
+@pytest.fixture(scope="module")
+def admin_tools_latest_track(juju: jubilant.Juju):
+    """Deploy temporal-admin-k8s from the latest track."""
+    deploy_temporal_stack(juju, temporal_admin_channel="latest/stable")
```

Comment:
Not a blocker: I'd also make this channel a global var.

Reply:
Addressed in 5228ceec25b545d59649eb38d3ddca1db9c7ffb9